tests/rgen/Microsoft.Macios.Generator.Tests/Classes/Data/Trampolines/ExpectedAVFoundationTrampolinesTrampolines.cs

```diff
@@ -0,0 +1,1369 @@
+
+	/// <summary>This class bridges native block invocations that call into C#</summary>
+	static internal class SDAVAudioConverterInputHandler
+	{
+		[Preserve (Conditional = true)]
+		[UnmanagedCallersOnly]
+		[UserDelegateType (typeof (global::AVFoundation.AVAudioConverterInputHandler))]
+		internal static unsafe global::ObjCRuntime.NativeHandle Invoke (global::System.IntPtr block_ptr, uint inNumberOfPackets, global::System.IntPtr* outStatus)
+		{
+			*outStatus = default;
+			var del = global::ObjCRuntime.BlockLiteral.GetTarget<global::AVFoundation.AVAudioConverterInputHandler> (block_ptr);
+			if (del is not null)
+			{
+				var ret = del (inNumberOfPackets, out global::System.Runtime.CompilerServices.Unsafe.AsRef<global::AVFoundation.AVAudioConverterInputStatus> (outStatus));
+				return global::ObjCRuntime.Runtime.RetainAndAutoreleaseNSObject (ret);
+			}
+			return default;
+		}
+
+		internal static unsafe global::ObjCRuntime.BlockLiteral CreateNullableBlock (global::AVFoundation.AVAudioConverterInputHandler? callback)
+		{
+			if (callback is null)
+				return default (global::ObjCRuntime.BlockLiteral);
+			return CreateBlock (callback);
+		}
+
+		[BindingImpl (BindingImplOptions.GeneratedCode | BindingImplOptions.Optimizable)]
+		internal static unsafe global::ObjCRuntime.BlockLiteral CreateBlock (global::AVFoundation.AVAudioConverterInputHandler callback)
+		{
+			delegate* unmanaged<global::System.IntPtr, uint, global::System.IntPtr*, global::ObjCRuntime.NativeHandle> trampoline = &Invoke;
+			return new global::ObjCRuntime.BlockLiteral (trampoline, callback, typeof (SDAVAudioConverterInputHandler), nameof (Invoke));
+		}
+	}
+	// TODO: generate trampoline for AVFoundation.AVAudioConverterInputHandler
+
+	[UnmanagedFunctionPointerAttribute (CallingConvention.Cdecl)]
+	[UserDelegateType (typeof (global::AVFoundation.AVAudioEngineManualRenderingBlock))]
+	unsafe internal delegate global::System.IntPtr DAVAudioEngineManualRenderingBlock (global::System.IntPtr block_ptr, uint numberOfFrames, global::ObjCRuntime.NativeHandle outBuffer, int* outError);
+
+	/// <summary>This class bridges native block invocations that call into C#</summary>
+	static internal class SDAVAudioEngineManualRenderingBlock
+	{
+		[Preserve (Conditional = true)]
+		[UnmanagedCallersOnly]
+		[UserDelegateType (typeof (global::AVFoundation.AVAudioEngineManualRenderingBlock))]
+		internal static unsafe global::System.IntPtr Invoke (global::System.IntPtr block_ptr, uint numberOfFrames, global::ObjCRuntime.NativeHandle outBuffer, int* outError)
+		{
+			*outError = default;
+			var del = global::ObjCRuntime.BlockLiteral.GetTarget<global::AVFoundation.AVAudioEngineManualRenderingBlock> (block_ptr);
+			if (del is not null)
+			{
+				var ret = del (numberOfFrames, new global::AudioToolbox.AudioBuffers (outBuffer), ref global::System.Runtime.CompilerServices.Unsafe.AsRef<int> (outError));
+				return (IntPtr) (long) ret;
+			}
+			return default;
+		}
```

Comment:
bgen code:
```sharp
internal static unsafe IntPtr Invoke (IntPtr block, uint numberOfFrames, NativeHandle outBuffer, int* outError) {
	var del = BlockLiteral.GetTarget<global::AVFoundation.AVAudioEngineManualRenderingBlock> (block);
	var retval = del (numberOfFrames, new global::AudioToolbox.AudioBuffers (outBuffer), ref global::System.Runtime.CompilerServices.Unsafe.AsRef<int> (outError));
	return (IntPtr) (long) retval;
}
```
Similar issue:

1. We init outError before the call.
2. We check for del no to be null. 

The rest of the code is the same.